### ISSUE TITLE
MethodWriter compile error if the same interface shows up multiple times in hierarchy

### ISSUE THREAD

**Issue by JerryShea** [OpenHFT/Chronicle-Wire#215]
See `GenerateMethodWriterInheritanceTest`

**Comment by glukos** [comment]
@JerryShea I've checked your test, the issue is easy to fix.
However, the test still fails as interfaces are declared with `private` modifier:
```
/net/openhft/chronicle/wire/method/GenerateMethodWriterInheritanceTestAnInterfaceGenerateMethodWriterInheritanceTestADescendantBinarylightMethodWriter.java:14: error: net.openhft.chronicle.wire.method.GenerateMethodWriterInheritanceTest.AnInterface has private access in net.openhft.chronicle.wire.method.GenerateMethodWriterInheritanceTest
public final class GenerateMethodWriterInheritanceTestAnInterfaceGenerateMethodWriterInheritanceTestADescendantBinarylightMethodWriter implements net.openhft.chronicle.wire.method.GenerateMethodWriterInheritanceTest.AnInterface, net.openhft.chronicle.wire.method.GenerateMethodWriterInheritanceTest.ADescendant, MethodWriter {
```
I'll change the test, please let me know if `private` was intentional.

**Comment by JerryShea** [comment]
That's great thanks Ivan. And yes private was a mistake

**Comment by JerryShea** [comment]
Sorry @glukos, found another case for you :) - https://github.com/OpenHFT/Chronicle-Wire/commit/792f12ad9615ef1a2e76ba1b3125efe05ba0bdfa

We handle this scenario in MethodReader so should handle it in MethodWriter too.

Reopening as similar enough to put under the same issue.

**Comment by glukos** [comment]
@JerryShea Thanks!
Apprarently, this scenario wasn'r handled in generated method reader. Fixed that as well.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.67](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.67), [BOM-2.20.87](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.87)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
